conjur/credentials_data.py

```diff
@@ -24,20 +27,24 @@ def __init__(self, machine=None, login=None, password=None):
         # especially the 'Use API Keys' section
         self.api_key = password
 
-    @staticmethod
-    def remove_entry_from_file(netrc_data, netrc_path):
+    @classmethod
+    def remove_entry_from_file(cls, netrc_data, netrc_path):
         """
         Method that removes the described login entry from netrc
         """
-        # pylint: disable=import-outside-toplevel
-        import netrc
-
         netrc_obj = netrc.netrc(netrc_path)
         hosts = netrc_obj.hosts
         hosts.pop(netrc_data['machine'], None)
 
         netrc_obj = netrc.netrc(netrc_path)
         netrc_obj.hosts.pop(netrc_data['machine'], None)
+        cls.build_netrc_from_instance(netrc_obj)
+
+    @classmethod
+    def build_netrc_from_instance(cls, netrc_obj):
```

Comment:
why is it a class method and not a util method in the UTILS class?

Reply:
I say the opposite, why is it a class method and not an instance method?
I must say I'm not a big fan of this class structure.
Could be better encapsulated IMO.
As in, let the object handle itself internally, and only expose what you need.
This mixture of static/class methods and instance methods could be better designed.

Reply:
@abrahamko it is not an instance method because it has no use for 'self' here. I redesigned this class so that credentials_data will be a POJO and credentials_from_file is where all the data manipulation is happening (updating the netrc, removing netrc data)

Reply:
Sigal I was thinking this is a different class this indeed seems to be the place for this kind of function. sorry